src/main/java/de/tum/in/www1/artemis/web/rest/plagiarism/PlagiarismCaseResource.java

```diff
@@ -66,6 +69,31 @@ public ResponseEntity<List<PlagiarismCase>> getPlagiarismCasesForCourseForInstru
         return getPlagiarismCasesResponseEntity(plagiarismCases);
     }
 
+    /**
+     * Retrieves all plagiarism cases related to an exam for the instructor view.
+     *
+     * @param courseId the course id to which the exam belongs
+     * @param examId the id of the exam
+     * @return all plagiarism cases of the course
+     */
+    @GetMapping("courses/{courseId}/exams/{examId}/plagiarism-cases/for-instructor")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<List<PlagiarismCase>> getPlagiarismCasesForExamForInstructor(@PathVariable long courseId, @PathVariable long examId) {
+        log.debug("REST request to get all plagiarism cases for instructor in exam with id: {}", examId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        if (!authenticationCheckService.isAtLeastInstructorInCourse(course, userRepository.getUserWithGroupsAndAuthorities())) {
+            throw new AccessForbiddenException("Only instructors of this course have access to its plagiarism cases.");
+        }
+
+        var plagiarismCases = plagiarismCaseRepository.findByExamIdWithPlagiarismSubmissionsAndComparison(examId);
+        plagiarismCases.forEach(plagiarismCase -> {
+            if (plagiarismCase.getExercise().getExerciseGroup().getExam().getCourse().getId() != courseId) {
+                throw new AccessForbiddenException("Exercise with id " + plagiarismCase.getExercise().getId() + " is not related to the given course id " + courseId);
+            }
+        });
```

Comment:
Would it be enough here to check that the exam with the given `examId` belongs to the course? If the exam belongs to a course, then all of its exercises should as well, I assume.

Reply:
I modified it to only check the exam of the first exercise since others should be the same. How does it look like now?